internal/elasticsearch/client.go

```diff
@@ -104,6 +107,27 @@ func (c *Client) doRequest(ctx context.Context, method, endpoint string, body io
 	if err != nil {
 		return nil, err
 	}
+
+	if body != nil {
+		buf := new(bytes.Buffer)
+		_, err = buf.ReadFrom(body)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = body.Seek(0,io.SeekStart)
+		if err != nil {
+			return nil, err
+		}
+
+		h := sha256.New()
+		h.Write(buf.Bytes())
+		bs := hex.EncodeToString(h.Sum(nil))
```

Comment:
There's a [body-signing function in aws-sdk-go](https://github.com/aws/aws-sdk-go/blob/0ace1685ac01310a6c6c173fa8fdd0cd9e04a55d/aws/signer/v4/v4.go#L766-L787) which is broadly similar to this. You might want to consider copying it so that we can ensure we're not missing an edge case, but it's basically the same.

It seems like aws-sdk-go only adds the header [if the target service is S3](https://github.com/aws/aws-sdk-go/blob/0ace1685ac01310a6c6c173fa8fdd0cd9e04a55d/aws/signer/v4/v4.go#L696-L700), which feels like a bug. aws-sdk-go-v2 may well fix this, so we should re-evaluate the necessity of this code as part of that ticket.